projects/noloan/admin/src/main/java/hasadna/noloan/admin/app/SuggetionRecyclerAdapter.java

```diff
@@ -2,42 +2,100 @@
 
 import android.os.Handler;
 import android.os.Looper;
+
 import androidx.annotation.NonNull;
 import androidx.recyclerview.widget.RecyclerView;
+
+import android.util.Log;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
 import android.widget.Button;
 import android.widget.TextView;
 import android.widget.Toast;
 
-import hasadna.noloan.admin.app.firestore.FirestoreClient;
+import java.util.ArrayList;
+
+import hasadna.noloan.firebase.DbMessages;
+import hasadna.noloan.firebase.FirestoreClient;
 import hasadna.noloan.protobuf.SmsProto.SmsMessage;
 
 public class SuggetionRecyclerAdapter
     extends RecyclerView.Adapter<SuggetionRecyclerAdapter.RecyclerViewHolder> {
 
-  DbMessagesHolder DbMessages;
+  DbMessages dbMessages;
+
+  ArrayList<SmsMessage> messages;
 
   public SuggetionRecyclerAdapter() {
-    DbMessages = DbMessagesHolder.getInstance();
+    dbMessages = dbMessages.getInstance();
+    messages = new ArrayList<>();
+
+    ArrayList<SmsMessage> list = dbMessages.getMessages();
+    for (int i = 0; i < list.size(); i++) {
+      SmsMessage message = list.get(i);
+      if (!message.getApproved()) {
+        Log.d("!!!!!!!!!!!!!!!!!!", "bla");
+        this.messages.add(message);
+      }
+    }
```

Comment:
Perhaps use `foreach` instead? 

    for (SmsMessage sms: dbMessages.getMessages()){
      if(!sms.getApproved()) {
        this.messages.add(sms);
      }
    }

Reply:
I don't like use foreach, It is less preferment. but done.